### PR TITLE
Chore: Remove several instances of non-strict null usage

### PR DESCRIPTION
**Pull request by kaydelaney**
**What this PR does / why we need it**:
Removes several instances of non-strict null usage.

### CODE REVIEW

File: public/app/features/explore/QueryEditor.tsx [hunk 1]
```diff
@@ -45,13 +45,13 @@ export default class QueryEditor extends PureComponent<QueryEditorProps, any> {
         target,
         refresh: () => {
           setTimeout(() => {
-            this.props.onQueryChange(target);
-            this.props.onExecuteQuery();
+            this.props.onQueryChange?.(target);
```
Comment (aocenas) [inline public/app/features/explore/QueryEditor.tsx]: This looks a bit weird. How come this did not crash before? Seems like the code assumes these functions to be non optional so maybe we should change the props to be non optional?

File: public/app/features/explore/Table.tsx [hunk 2]
```diff
@@ -28,7 +28,7 @@ export default class Table extends PureComponent<TableProps> {
           if (link.className === 'link') {
             const columnKey = column.Header().props.title;
             const rowValue = rowInfo.row[columnKey];
-            this.props.onClickCell(columnKey, rowValue);
+            this.props.onClickCell?.(columnKey, rowValue);
```
Comment (aocenas) [inline public/app/features/explore/Table.tsx]: Same question whether this should not be non optional in props?

**Comment by aocenas** [comment]
Nice work! Btw typescript does not have anything like partial strict mode? Like marking these files as strict so they won't regress later on?

**Comment by torkelo** [comment]
Sadly no, it's not possible to set strict on a file basis :( 

We can lower this limit and have PRs fail if they go over it: https://github.com/grafana/grafana/blob/master/scripts/ci-frontend-metrics.sh#L6 

Really great to see this PR, we need find a way to get work like this into our weekly work so we can make significant progress and eventually enable strict mode.

File: public/app/features/explore/QueryEditor.tsx (hunk 1 above)
Comment (kaydelaney) [reply public/app/features/explore/QueryEditor.tsx]: Not sure, but [QueryEditorRow](https://github.com/grafana/grafana/blob/b756aa0bb197286963640ebb29501c82c04f3725/public/app/features/dashboard/panel_editor/QueryEditorRow.tsx#L153) doesn't seem to pass these props for example, so I'm not sure we can just make them non-optional.

File: public/app/features/explore/Table.tsx (hunk 2 above)
Comment (kaydelaney) [reply public/app/features/explore/Table.tsx]: Prop doesn't seem to be passed in [TablePanel](https://github.com/grafana/grafana/blob/9b7748ec133cb2ce4b092e523cb6d70a7b89de83/public/app/plugins/panel/table2/TablePanel.tsx#L26) so I think it's supposed to be optional?

**Review by Estrax — APPROVED** [review]
LGTM. Great work!

File: public/app/features/explore/QueryEditor.tsx (hunk 1 above)
Comment (aocenas) [reply public/app/features/explore/QueryEditor.tsx]: Ah ok in that case I assume the code before could crash in some cases.

File: public/app/features/explore/QueryEditor.tsx [hunk 3]
```diff
@@ -45,13 +45,13 @@ export default class QueryEditor extends PureComponent<QueryEditorProps, any> {
...
+            this.props.onExecuteQuery?.();
           }, 1);
         },
         onQueryChange: () => {
           setTimeout(() => {
-            this.props.onQueryChange(target);
+            this.props.onQueryChange?.(target);
```
Comment (torkelo) [inline public/app/features/explore/QueryEditor.tsx]: wonder why this prop is optional?

**Pull request merged**